lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -1078,14 +1078,29 @@ llvm::Error Caffe2ModelLoader::loadWeights(caffe2::NetDef &net) {
 Caffe2ModelLoader::Caffe2ModelLoader(const std::string &netDescFilename,
                                      const std::string &netWeightFilename,
                                      llvm::ArrayRef<const char *> names,
-                                     llvm::ArrayRef<TypeRef> types, Function &F)
-    : CommonOperatorLoader(names, types, F) {
-  // The caffe2 network descriptor that we are deserializing.
-  caffe2::NetDef networkDef = EXIT_ON_ERR(loadProtoFile(netDescFilename));
+                                     llvm::ArrayRef<TypeRef> types, Function &F,
+                                     llvm::Error *errPtr)
+    : CommonOperatorLoader(names, types, F, errPtr) {
+  // Lambda to setup the Caffe2ModelLoader and return any llvm::Errors that were
+  // raised.
+  auto setup = [&]() -> llvm::Error {
+    // The caffe2 network descriptor that we are deserializing.
+    caffe2::NetDef networkDef;
+    ASSIGN_VALUE_OR_RETURN_ERR(networkDef, loadProtoFile(netDescFilename));
+
+    // The caffe2 weights that we are deserializing.
+    caffe2::NetDef weightsDef;
+    ASSIGN_VALUE_OR_RETURN_ERR(weightsDef, loadProtoFile(netWeightFilename));
+
+    RETURN_IF_ERR(loadWeights(weightsDef));
+    RETURN_IF_ERR(loadNetwork(networkDef));
 
-  // The caffe2 weights that we are deserializing.
-  caffe2::NetDef weightsDef = EXIT_ON_ERR(loadProtoFile(netWeightFilename));
+    return llvm::Error::success();
```

Comment:
I found `Error::success()` a bit counter-intuitive. Is this standard practice in LLVM?

Reply:
Yes this is the llvm standard practice as far as I can tell. It is just a special error value that signals no error has occurred. I also found this somewhat counterintuitive so I originally defined a RETURN_SUCCESS() macro to try to make it a bit more readable but this was removed during a previous code review.